content/2016.12.how-to-develop-a-webapp-with-Kinto.md

```diff
@@ -0,0 +1,256 @@
+
+```js
+// https://discute.re/#4815162342  is mapped to the collection 4815162342
+this.collectionName = window.location.hash.slice(1);
+```
+
+Then we need to do a GET request to get the list of all room's messages.
+A simple `GET https://kinto.discute.re/v1/buckets/chat/collections/4815162342/records` would do the job (with fetch or XMLHttpRequest), but we will rather be using the [kinto-https-js](https://github.com/kinto/kinto-http.js) library that offers a clean client API, take a look at this code:
+
+    const kinto_url = 'https://kinto.discute.re/v1';
+    // options stores the connection identifiers
+    const client = new KintoClient(kinto_url, options);
+    client.bucket('chat')
+        .collection('4815162342')
+        .listRecords()
+        .then(({data}) => displayMessage(data));
+
+The `options` variable contains at least the authentication token. This token was not created before. Kinto accepts all requests and responds accordingly to the token: if the token gives access to the resource, the user can act on it (create, update, etc.), otherwise he can't.
+[Read more about authentication here](https://kinto.readthedocs.io/en/stable/api/1.x/authentication.html).
+
+`displayMessage(data)` can now retrieve the JSON object and render it as HTML.
+
+```json
+[
+    {
+        "message": "I know nothing",
+        "last_modified": 1480340246147,
+        "author": "Jon Snow",
+        "id": "a1e6e879-5c94-4a01-9ec2-28e152d1d8f4"
+    },
+    {
+        "message": "I know that I know nothing",
+        "last_modified": 1480340396521,
+        "author": "Socrates",
+        "id": "94787c48-aa96-4121-879f-1bd68d3c1f23"
+    }
+]
+```
+
+```js
+getMessage(message, author) {
+    return (
+    `<li>
+      <span>
+        <span>${author}</span>
+        <span>${message}</span>
+      </span>
+    </li>`
+    );
+}
+
+displayMessage({message, author}) {
+    var container = document.createElement('div');
+    container.innerHTML = this.getMessage(message, author);
+    this.messagesList.firstChild.appendChild(container.firstChild);
+}
+```
+
+Here is how the HTML code looks like:
+
+```html
+<div id="messages">
+  <ul>
+    <li>
+      <span>
+        <span>Jon Snow</span>
+        <span>I know nothing</span>
+      </span>
+    </li>
+    <li>
+      <span>
+        <span>Socrates</span>
+        <span>I know that I know nothing</span>
+      </span>
+    </li>
+  </ul>
+</div>
+```
+
+## Post a new message
+
+This time, we only have to send the form's data to the Kinto server.
+
+![](images/kintochat-new-post.png)
+{.image}
+
+Once again, we could go with a simple POST request `https://kinto.discute.re/v1/buckets/chat/collections/4815162342/records` but here we will use again the Kinto client library to write:
+
+```js
+const messageInput = document.getElementById('message');
+const messageForm = document.getElementById('message-form');
```

Comment:
I realize that the form in question doesn't appear anywhere in our code snippets; we may want to showcase it in the code block above